core/capabilities/ccip/ocrimpls/svm_contract_transmitter_factory.go

```diff
@@ -0,0 +1,180 @@
+package ocrimpls
+
+import (
+	"fmt"
+
+	"github.com/smartcontractkit/libocr/offchainreporting2plus/ocr3types"
+	ocrtypes "github.com/smartcontractkit/libocr/offchainreporting2plus/types"
+
+	"github.com/smartcontractkit/chainlink-ccip/pkg/consts"
+	"github.com/smartcontractkit/chainlink-ccip/pkg/types/ccipocr3"
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+	"github.com/smartcontractkit/chainlink/v2/core/capabilities/ccip/ccipevm"
+	"github.com/smartcontractkit/chainlink/v2/core/capabilities/ccip/ccipsolana"
+
+	ccipcommon "github.com/smartcontractkit/chainlink/v2/core/capabilities/ccip/common"
+)
+
+// SVMCommitCallArgs defines the calldata structure for an SVM commit transaction.
+// IMPORTANT: The names and types of the fields are critical because the chainwriter uses mapstructure
+// to map these fields to the contract's parameter names. Changing these names or types (or omitting the
+// mapstructure tags) may result in transactions being constructed with incorrect arguments.
+type SVMCommitCallArgs struct {
+	ReportContext [2][32]byte               `mapstructure:"ReportContext"`
+	Report        []byte                    `mapstructure:"Report"`
+	Rs            [][32]byte                `mapstructure:"Rs"`
+	Ss            [][32]byte                `mapstructure:"Ss"`
+	RawVs         [32]byte                  `mapstructure:"RawVs"`
+	Info          ccipocr3.CommitReportInfo `mapstructure:"Info"`
+}
+
+// SVMExecCallArgs defines the calldata structure for an SVM execute transaction.
+// IMPORTANT: The names and types of the fields are critical because the chainwriter uses mapstructure
+// to map these fields to the contract's parameter names. Changing these names or types (or omitting the
+// mapstructure tags) may result in transactions being constructed with incorrect arguments.
+type SVMExecCallArgs struct {
```

Comment:
Could we move these types to a common place so it can be imported into the solana repo?

Reply:
Which repo did you have in mind? Could we import this from the Solana repo instead of defining it here?

Reply:
I don't have a preference, was thinking `chainlink-ccip` originally but Solana repo is fine too